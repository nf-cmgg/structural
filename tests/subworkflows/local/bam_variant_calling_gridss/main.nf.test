nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_GRIDSS"
    script "subworkflows/local/bam_variant_calling_gridss/main.nf"
    workflow "BAM_VARIANT_CALLING_GRIDSS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_gridss"

    config "./nextflow.config"

    test("homo_sapiens") {

        setup {
            run("UNTAR") {
                script "modules/nf-core/untar/main.nf"
                process {
                    """
                    input[0] = Channel.value([
                        [id:"bwa"],
                        file(params.bwa, checkIfExists: true)
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = UNTAR.out.untar
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.gridss_vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.gridss_vcfs[0][2]).name
                ).match() }
            )
        }

    }

}
