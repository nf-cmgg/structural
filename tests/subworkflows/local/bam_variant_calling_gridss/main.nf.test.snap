{
    "homo_sapiens": {
        "content": [
            [
                "chr16\t1549720\tgridss_83\tT\t.TCACAACTCTAATCTTACAAGGTAATTAAGTAATTATTATCACCCACTCCCTCT\t176.89\tLOW_QUAL;NO_RP\tSVLEN=0;ANRP=0;ANRPQ=0;ANSR=0;ANSRQ=0;AS=0;ASC=1X;ASQ=0;ASRP=0;ASSR=0;BA=1;BAQ=88.45;BASRP=0;BASSR=4;BEID=asm0-2472;BEIDH=-1;BEIDL=53;BMQ=60;BMQN=60;BMQX=60;BQ=176.89;BSC=4;BSCQ=88.45;BUM=0;BUMQ=0;BVF=4;CAS=0;CASQ=0;CIPOS=0,0;CQ=176.89;EVENT=gridss0b_947;IC=0;IQ=0;RAS=0;RASQ=0;REF=24;REFPAIR=3;RP=0;RPQ=0;SB=1;SC=1X;SR=0;SRQ=0;SVTYPE=BND;VF=0;ORGBEID=gridss0b_947b\tGT:AF:ANRP:ANRPQ:ANSR:ANSRQ:ASQ:ASRP:ASSR:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF\t0/0:0.129:0:0:0:0:0:0:0:88.45:0:4:176.89:4:88.45:0:0:4:0:0:0:0:0:24:3:0:0:0:0:0",
                "chr16\t1554547\tgridss_84\tA\t.TTTACCCCTTTTTGCAGTTTTTCTGTTAGATCATATTTGCAGAAATGAACACCACCCTGGCCAACCAAAAAAAAATTTTTATAA\t124.89\tLOW_QUAL;NO_RP\tSVLEN=0;ANRP=0;ANRPQ=0;ANSR=0;ANSRQ=0;AS=0;ASC=1X;ASQ=0;ASRP=0;ASSR=0;BA=1;BAQ=62.44;BASRP=0;BASSR=3;BEID=asm0-2487;BEIDH=-1;BEIDL=83;BMQ=60;BMQN=60;BMQX=60;BQ=124.89;BSC=3;BSCQ=62.44;BUM=0;BUMQ=0;BVF=3;CAS=0;CASQ=0;CIPOS=0,0;CQ=124.89;EVENT=gridss0b_965;IC=0;IQ=0;RAS=0;RASQ=0;REF=28;REFPAIR=5;RP=0;RPQ=0;SB=0;SC=1X;SR=0;SRQ=0;SVTYPE=BND;VF=0;ORGBEID=gridss0b_965b\tGT:AF:ANRP:ANRPQ:ANSR:ANSRQ:ASQ:ASRP:ASSR:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF\t0/0:0.083:0:0:0:0:0:0:0:62.44:0:3:124.89:3:62.44:0:0:3:0:0:0:0:0:28:5:0:0:0:0:0",
                "chr16\t1574798\tgridss_85\tG\t.ATATACTGATGAGGCCGGACACGGGGGCTCACCCCTGTAATCCAGCACTTTGGGAGGCGGAGGCGG\t422.64\tLOW_QUAL;NO_RP\tSVLEN=0;ANRP=0;ANRPQ=0;ANSR=0;ANSRQ=0;AS=0;ASC=1X;ASQ=0;ASRP=0;ASSR=0;BA=1;BAQ=221.64;BASRP=0;BASSR=11;BEALN=chr16:1353495|+|25S40M|0;BEID=asm0-2524;BEIDH=-1;BEIDL=65;BMQ=58.18;BMQN=40;BMQX=60;BQ=422.64;BSC=10;BSCQ=201;BUM=0;BUMQ=0;BVF=11;CAS=0;CASQ=0;CIPOS=0,0;CQ=422.64;EVENT=gridss0b_991;IC=0;IQ=0;RAS=0;RASQ=0;REF=6;REFPAIR=2;RP=0;RPQ=0;SB=0.714286;SC=1X;SR=0;SRQ=0;SVTYPE=BND;VF=0;ORGBEID=gridss0b_991b\tGT:AF:ANRP:ANRPQ:ANSR:ANSRQ:ASQ:ASRP:ASSR:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF\t0/1:0.579:0:0:0:0:0:0:0:221.64:0:11:422.64:10:201:0:0:11:0:0:0:0:0:6:2:0:0:0:0:0",
                "chr16\t1574808\tgridss_86\tC\tCCCTGTAATCCAGCACTTTGGGAGGCGGAGGCGGGTGGATCACCTGAGGTCAGGAGTTCAAGA.\t204.38\tLOW_QUAL;NO_RP\tSVLEN=0;ANRP=0;ANRPQ=0;ANSR=0;ANSRQ=0;AS=0;ASC=1X;ASQ=0;ASRP=0;ASSR=0;BA=1;BAQ=102.19;BASRP=0;BASSR=5;BEALN=chrX:1120041|-|52M1D10M|0;BEID=asm0-1247;BEIDH=-1;BEIDL=0;BMQ=60;BMQN=60;BMQX=60;BQ=204.38;BSC=5;BSCQ=102.19;BUM=0;BUMQ=0;BVF=5;CAS=0;CASQ=0;CIPOS=0,0;CQ=204.38;EVENT=gridss0f_1036;IC=0;IQ=0;RAS=0;RASQ=0;REF=12;REFPAIR=3;RP=0;RPQ=0;SB=0.6;SC=1X;SR=0;SRQ=0;SVTYPE=BND;VF=0;ORGBEID=gridss0f_1036b\tGT:AF:ANRP:ANRPQ:ANSR:ANSRQ:ASQ:ASRP:ASSR:BAQ:BASRP:BASSR:BQ:BSC:BSCQ:BUM:BUMQ:BVF:CASQ:IC:IQ:QUAL:RASQ:REF:REFPAIR:RP:RPQ:SR:SRQ:VF\t0/0:0.25:0:0:0:0:0:0:0:102.19:0:5:204.38:5:102.19:0:0:5:0:0:0:0:0:12:3:0:0:0:0:0"
            ],
            "test.gridss.vcf.gz.tbi",
            5,
            0
        ],
        "timestamp": "2023-11-29T16:41:25.529471036"
    }
}